src/charm_helpers.py

```diff
@@ -0,0 +1,169 @@
+        "STORAGE_BUCKET_STATE": config["storage-bucket-state"],
+        "STORAGE_BUCKET_WORKLOAD_OUTPUT": config["storage-bucket-workload-output"],
+        "STORAGE_BUCKET_ACTIVITY_PAYLOAD": config["storage-bucket-activity-payload"],
+        "CONFIGS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.23.002",
+        "JOBS_DATABASE_MINIMUM_FLYWAY_MIGRATION_VERSION": "0.40.26.001",
+        "MICRONAUT_ENVIRONMENTS": "control-plane",
+        "WORKERS_MICRONAUT_ENVIRONMENTS": "control-plane",
+        "CRON_MICRONAUT_ENVIRONMENTS": "control-plane",
+        "INTERNAL_API_HOST": f"localhost:{INTERNAL_API_PORT}",
+        "CONNECTOR_BUILDER_SERVER_API_HOST": f"localhost:{CONNECTOR_BUILDER_SERVER_API_PORT}",
+        "AIRBYTE_API_HOST": f"localhost:{AIRBYTE_API_PORT}",
+        "WORKLOAD_API_HOST": f"localhost:{WORKLOAD_API_PORT}",
+        "WORKLOAD_API_URL": f"localhost:{WORKLOAD_API_PORT}",
+        "LOG_LEVEL": config["log-level"].value,
+        "MICROMETER_METRICS_ENABLED": "false",
+        "LAUNCHER_MICRONAUT_ENVIRONMENTS": "control-plane,oss",
+        "KEYCLOAK_INTERNAL_HOST": "localhost",
+        "KEYCLOAK_DATABASE_URL": db_url + "?currentSchema=keycloak",
+        "WEBAPP_URL": "airbyte-ui-k8s:8080",
+        "WORKER_ENVIRONMENT": "kubernetes",
+        "SECRET_PERSISTENCE": "TESTING_CONFIG_DB_TABLE",
+        "SHOULD_RUN_NOTIFY_WORKFLOWS": "true",
+        "CONNECTOR_BUILDER_API_URL": "/connector-builder-api",
+        "TEMPORAL_WORKER_PORTS": "9001,9002,9003,9004,9005,9006,9007,9008,9009,9010,9011,9012,9013,9014,9015,9016,9017,9018,9019,9020,9021,9022,9023,9024,9025,9026,9027,9028,9029,9030",
+        "JOB_KUBE_SERVICEACCOUNT": app_name,
+        "JOB_KUBE_NAMESPACE": model_name,
+        "RUNNING_TTL_MINUTES": config["pod-running-ttl-minutes"],
+        "SUCCEEDED_TTL_MINUTES": config["pod-successful-ttl-minutes"],
+        "UNSUCCESSFUL_TTL_MINUTES": config["pod-unsuccessful-ttl-minutes"],
+    }
+
+    if config["storage-type"].value == StorageType.minio and state.minio:
+        env.update(
+            {
+                "MINIO_ENDPOINT": minio_endpoint,
+                "AWS_ACCESS_KEY_ID": state.minio["access-key"],
+                "AWS_SECRET_ACCESS_KEY": state.minio["secret-key"],
+                "S3_PATH_STYLE_ACCESS": "true",
+            }
+        )
+
+    if config["storage-type"].value == StorageType.s3 and state.s3:
+        env.update(
+            {
+                "AWS_ACCESS_KEY_ID": state.s3["access-key"],
+                "AWS_SECRET_ACCESS_KEY": state.s3["secret-key"],
+                "S3_LOG_BUCKET_REGION": state.s3["region"],
+                "AWS_DEFAULT_REGION": state.s3["region"],
+            }
+        )
+
+    http_proxy = os.environ.get("JUJU_CHARM_HTTP_PROXY")
+    https_proxy = os.environ.get("JUJU_CHARM_HTTPS_PROXY")
+    no_proxy = os.environ.get("JUJU_CHARM_NO_PROXY")
+    java_tool_options = _get_java_tool_options(http_proxy, https_proxy, no_proxy)
+
+    if http_proxy or https_proxy:
```

Comment:
question (non-blocking): What happens if one is `None` and you put `None` for that proxy variable below?

Would it make sense to start with an empty dictionary and only update environment variables for valid proxies?

Reply:
Makes more sense, updated accordingly to check each and update the env based on that.